queries/course_requests.js

```diff
@@ -1,31 +1,92 @@
 /**
- * This file holds the code of all queries related to the CourseRequests table.
- *
+ * This file holds the code of all queries related to the courseRequests table.
+ * 
  * @version 13/09/2022
  */
+
+const helperFunctions = require("../helper_functions.js");
+
 module.exports = {
-  /**
-   * Get the total number of course requests in the database.
-   *
-   * A status of 200 is given for successful requests. A status of 400 for unsuccessful ones.
-   *
-   * @param {*} sql The mssql instance connected to the database currently used by the API.
-   * @param {*} res The object to send result to a given query sender.
-   */
-  getNumberOfCourseRequests: function (sql, res) {
+
+    /**
+     * Retrieves all the new course requests and update their status to 1 (= pending) if the retrieval is successful.
+     * 
+     * @param {*} sql The mssql instance connected to the database currently used by the API.
+     * @param {*} res The object to send result to a given query sender.
+     */
+    getNewCourseRequests: function (sql, res) {
+
+        const request = new sql.Request();
+
+        request
+            .query('SELECT Subject, Frequency, Levels.Name AS LevelName, Levels.CostPerHour*CourseRequests.Duration AS Money, Duration, Day, Time FROM CourseRequests JOIN DateOptions ON CourseRequests.ID = DateOptions.CourseRequestID JOIN Levels ON CourseRequests.LevelID = Levels.ID where Status = 0', function (err, recordset) {
+                if (err) {
+                    console.log(err);
+                    res.status(400).json({
+                        error: err
+                    });
+                } else {
+
+                    res.status(200).json({
+                        "result": helperFunctions.arrangeResult(recordset.recordset)
+                    });
+
+                    updateCourseRequests(sql);
+                }
+            });
+    },
+
+    /**
+     * Get the total number of course requests in the database.
+     *
+     * A status of 200 is given for successful requests. A status of 400 for unsuccessful ones.
+     *
+     * @param {*} sql The mssql instance connected to the database currently used by the API.
+     * @param {*} res The object to send result to a given query sender.
+     */
+    getNumberOfCourseRequests: function (sql, res) {
+        const request = new sql.Request();
+
+        request.query(
+            "SELECT COUNT(ID) FROM CourseRequests WHERE Status=0",
+            function (err, recordset) {
+                if (err) {
+                    res.status(400).json({
+                        error: err
+                    });
+                }
+
+                res
```

Comment:
Probably not a good idea to send request to the caller in that function. The caller does not and should not know act the updating and this can maybe cause issues

Reply:
I mean I wasn't responsible for doing that function. I just pulled main.